### PR TITLE
Factorial function minor changes

### PR DESCRIPTION
**Pull request by JM-299792458**
Thought of using an array to store numbers and to compute factorial using DP but that would have increased space complexity.

### CODE REVIEW

File: lib/libc/math.c [hunk 1]
```diff
@@ -39,17 +39,16 @@ pow(double base, double power)
...
-    double i;
-    double return_value = 1;
-
-    for (i = 1; i < number; i++)
-    {
-        return_value *= i;
-    }
-
-    return return_value;
+factorial(double number) {
```
Comment (Keyboard-Slayer) [inline lib/libc/math.c:42]: thank you very much for taking the time to contribute to the projects.

I know that I haven't written the coding style guide yet :sweat_smile:
But there is a lot of issue coding style wise

Here's a few examples of what you should do: 
- Declaring **ALL** the variable at the beginning of the function (even for the `for` statement)
- Adding a new line after each `{`, `}` and keywords (such as `if`, `else`, ...)
- Adding a new line before the `return` 
- Running `make format` before submitting your code

**Review by Keyboard-Slayer — CHANGES_REQUESTED** [review]

File: lib/libc/math.c (hunk 1 above)
Comment (JM-299792458) [reply lib/libc/math.c:42]: Okay.

File: lib/libc/math.c [hunk 2]
```diff
@@ -39,17 +39,26 @@ pow(double base, double power)
...
+    
+    if (number == 0 || number == 1) 
+    {
         return 1;
-    } else {
-        double i;
-    	double result = 1;
-    	for (i = 2; i <= number; i++) {
+    }
+
```
Comment (Keyboard-Slayer) [inline lib/libc/math.c:51]: You should just remove the space here

File: lib/libc/math.c [hunk 3]
```diff
@@ -39,17 +39,26 @@ pow(double base, double power)
...
+    else 
+    {
+    	for (i = 2; i <= number; i++) 
+        {
         	result *= i;
     	}
+
     	return result;
 	}
```
Comment (Keyboard-Slayer) [inline lib/libc/math.c:60]: Align that correctly and it'll be perfect

**Review by Keyboard-Slayer — CHANGES_REQUESTED** [review]

**Review by d0p1s4m4 — APPROVED** [review]

**Comment by Keyboard-Slayer** [comment]
LGTM :+1:

**Pull request merged**